src/type/__tests__/introspection-test.ts

```diff
@@ -1519,6 +1536,105 @@ describe('Introspection', () => {
     });
   });
 
+  it('identifies oneOf for input objects', () => {
+    const schema = buildSchema(`
+      input SomeInputObject @oneOf {
+        a: String
+      }
+
+      input AnotherInputObject {
+        a: String
+        b: String
+      }
+
+      type Query {
+        someField(someArg: SomeInputObject): String
+        anotherField(anotherArg: AnotherInputObject): String
+      }
+    `);
+
+    const source = `
+      {
+        a: __type(name: "SomeInputObject") {
+          isOneOf
+        }
+        b: __type(name: "AnotherInputObject") {
+          isOneOf
+        }
+      }
+    `;
+
+    expect(graphqlSync({ schema, source })).to.deep.equal({
+      data: {
+        a: {
+          isOneOf: true,
+        },
+        b: {
+          isOneOf: false,
+        },
+      },
+    });
+  });
+
+  it('returns null for oneOf for other types', () => {
+    const schema = buildSchema(`
+      type A implements I {
+        a: String
+      }
+      enum E {
+        A
+      }
+      interface I {
+        a: String
+      }
+      union U = A
+      type Query {
+        someField(e: E): U
+        anotherField(e: E): I
+      }
+    `);
+
+    const source = `
+      {
+        a: __type(name: "A") {
+          isOneOf
+        }
+        e: __type(name: "E") {
+          isOneOf
+        }
+        i: __type(name: "I") {
+          isOneOf
+        }
+        o: __type(name: "String") {
+          isOneOf
+        }
+        u: __type(name: "U") {
+          isOneOf
+        }
+      }
+    `;
+
+    expect(graphqlSync({ schema, source })).to.deep.equal({
+      data: {
+        a: {
```

Comment:
nit: In this and previous test you can shorten it by doing
```suggestion
        a: { isOneOf: null },
```
as a single line